File: rust/armstrong_numbers.rs
pub fn is_armstrong_number(num: u32) -> bool {
    let digits = num.to_string().len() as u32;

    let mut n = num;
    let mut sum = 0;
    while n > 0 {
        let d = n % 10;
        let t = d.pow(digits);
        if u32::MAX - t < sum {
            return false;
        }
        sum += t;
        n /= 10;
    }

    sum == num
}

File: zig/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    var len: u128 = 0;
    var n = num;
    while (n > 0) : (len += 1) {
        n /= 10;
    }
    n = num;
    var sum: u128 = 0;
    while (n > 0) : (n /= 10) {
        sum += std.math.pow(u128, n % 10, len);
    }
    return sum == num;
}
